apps/archive/common.py

```diff
@@ -502,16 +505,23 @@ def remove_media_files(doc):
                 )
 
                 if references.count() == 0:
-                    logger.info("Deleting media:{}".format(rendition.get("media")))
+                    logger.info("Deleting media:%s", media)
                     app.media.delete(media)
+                else:
+                    logger.info("Keeping media:%s due to references", media)
             except Exception:
-                logger.exception("Failed to remove Media Id: {} from item: {}".format(media, doc.get(config.ID_FIELD)))
+                logger.exception("Failed to remove Media Id: %s from item: %s", media, doc.get("guid"))
 
     for attachment in doc.get("attachments", []):
         lookup = {"_id": attachment["attachment"]}
         get_resource_service("attachments").delete_action(lookup)
 
 
+def remove_media_references(item_id, published):
+    get_resource_service("media_references").delete_action({"item_id": item_id, "published": published})
```

Comment:
Would this need an index for `published`, or do you think this should be fast enough?

Reply:
there is `item_id` index and then it's only a few items to be checked so should be fine